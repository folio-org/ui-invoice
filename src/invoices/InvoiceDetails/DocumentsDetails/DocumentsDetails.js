import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import {
  Col,
  Row,
  Label,
  MultiColumnList,
} from '@folio/stripes/components';

import InvoiceDocument from '../../InvoiceDocument';

const linksVisibleColumns = ['name', 'url'];
const linksColumnMapping = {
  name: <FormattedMessage id="ui-invoice.invoice.link.name" />,
  url: <FormattedMessage id="ui-invoice.invoice.link.url" />,
};
const linksFormatter = {
  url: l => (
    <a
      href={l.url}
      target="_blank"
      rel="noopener noreferrer"
    >
      {l.url}
    </a>
  ),
};

const documentsVisibleColumns = ['name'];
const documentsColumnMapping = {
  name: <FormattedMessage id="ui-invoice.invoice.documents.name" />,
};
const documentsFormatter = {
  name: d => (
    <InvoiceDocument
      name={d.name}
      documentId={d.id}
    />
  ),
};

const DocumentsDetails = ({ invoiceDocuments }) => {
  return (
    <>
      <Row>
        <Col xs={12}>
          <Label>
            <FormattedMessage id="ui-invoice.invoice.links.title" />
          </Label>

          <MultiColumnList
            contentData={invoiceDocuments.filter(invoiceDocument => invoiceDocument.url)}
            visibleColumns={linksVisibleColumns}
            columnMapping={linksColumnMapping}
            formatter={linksFormatter}
            interactive={false}
          />
        </Col>
      </Row>

      <Row>
        <Col xs={12}>
          <Label>
            <FormattedMessage id="ui-invoice.invoice.documents.title" />
          </Label>

          <MultiColumnList
            contentData={invoiceDocuments.filter(invoiceDocument => !invoiceDocument.url)}
            visibleColumns={documentsVisibleColumns}
            columnMapping={documentsColumnMapping}
            formatter={documentsFormatter}
            interactive={false}
          />
        </Col>
      </Row>
    </>
  );
};

DocumentsDetails.propTypes = {
  invoiceDocuments: PropTypes.arrayOf(PropTypes.object),
};

DocumentsDetails.defaultProps = {
  invoiceDocuments: [],
};

export default DocumentsDetails;
