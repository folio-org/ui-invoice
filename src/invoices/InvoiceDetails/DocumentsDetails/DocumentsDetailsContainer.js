import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { get } from 'lodash';

import { stripesConnect } from '@folio/stripes/core';

import { invoiceDocumentsResource } from '../../../common/resources';
import DocumentsDetails from './DocumentsDetails';

export const DocumentsDetailsContainer = ({ resources }) => {
  const invoiceDocuments = get(resources, 'invoiceDocumentsDetails.records', []);

  return <DocumentsDetails invoiceDocuments={invoiceDocuments} />;
};

DocumentsDetailsContainer.manifest = Object.freeze({
  invoiceDocumentsDetails: invoiceDocumentsResource,
});

DocumentsDetailsContainer.propTypes = {
  resources: PropTypes.object.isRequired,
};

export default withRouter(stripesConnect(DocumentsDetailsContainer));
