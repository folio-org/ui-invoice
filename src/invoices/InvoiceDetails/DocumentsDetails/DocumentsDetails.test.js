import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import DocumentsDetails from './DocumentsDetails';

const defaultProps = {
  invoiceDocuments: [
    {
      id: 'documentId1',
      name: 'ABA incoice',
      link: 'https://folio.com/document',
    },
    {
      id: 'documentId2',
      name: 'ABA voucher',
      link: 'https://folio.com/voucher',
    },
  ],
};
const renderDocumentsDetails = (props = defaultProps) => render(
  <DocumentsDetails {...props} />,
  { wrapper: MemoryRouter },
);

describe('DocumentsDetails', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render invoice documents', () => {
    const { asFragment } = renderDocumentsDetails();

    expect(asFragment()).toMatchSnapshot();
  });
});
