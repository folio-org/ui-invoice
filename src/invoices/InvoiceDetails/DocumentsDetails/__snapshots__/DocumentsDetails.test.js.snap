// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentsDetails should render invoice documents 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col-xs-12"
    >
      <label
        class="label"
      >
        ui-invoice.invoice.links.title
      </label>
      <div
        class="mclEmptyMessage"
        style="min-width: 200px;"
        tabindex="0"
      >
        stripes-components.tableEmpty
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-xs-12"
    >
      <label
        class="label"
      >
        ui-invoice.invoice.documents.title
      </label>
      <div
        aria-live="assertive"
        aria-relevant="additions"
        class="sr-only"
        tabindex="-1"
      />
      <div
        aria-rowcount="3"
        class="mclContainer"
        data-total-count="0"
        role="grid"
        style="position: relative; overflow: hidden; height: auto; width: 100%;"
        tabindex="0"
      >
        <div
          class="mclHeaderContainer"
        >
          <div
            aria-rowindex="1"
            class="mclHeaderRow"
            role="row"
            style="display: flex;"
          >
            <div
              aria-sort="none"
              class="mclHeader"
              id="list-column-name"
              role="columnheader"
              style="flex: 0 0 auto;"
            >
              ui-invoice.invoice.documents.name
            </div>
          </div>
        </div>
        <div
          class="mclScrollable"
          style="position: static; width: 100%;"
        >
          <div
            class="mclRowContainer"
            role="rowgroup"
            style="position: static;"
          >
            <div
              aria-rowindex="2"
              class="mclRowFormatterContainer"
              data-row-index="row-0"
              role="row"
              style="position: static;"
            >
              <div
                class="mclRow mclIsOdd"
                data-row-inner="0"
                style="min-width: 0px;"
              >
                <div
                  class="mclCell  mclCellStyle"
                  role="gridcell"
                >
                  <button
                    class="invoiceDocumentButton"
                    type="button"
                  >
                    ABA incoice
                  </button>
                </div>
              </div>
            </div>
            <div
              aria-rowindex="3"
              class="mclRowFormatterContainer"
              data-row-index="row-1"
              role="row"
              style="position: static;"
            >
              <div
                class="mclRow"
                data-row-inner="1"
                style="min-width: 0px;"
              >
                <div
                  class="mclCell  mclCellStyle"
                  role="gridcell"
                >
                  <button
                    class="invoiceDocumentButton"
                    type="button"
                  >
                    ABA voucher
                  </button>
                </div>
              </div>
            </div>
            <div
              class="mclEndOfListContainer"
              data-end-of-list="2"
            >
              <div
                class="mclCenteredContainer"
                style="width: 100%; visibility: hidden; height: 0px; padding: 0px;"
              >
                <div
                  class="textCentered"
                  data-test-layout="true"
                >
                  <div
                    class="mclEndOfList"
                    data-test-layout="true"
                  >
                    <span>
                      Icon
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
